src/validation.cpp

```diff
@@ -995,16 +1015,28 @@ bool MemPoolAccept::ReplacementChecks(Workspace& ws)
         // Even though this is a fee-related failure, this result is TX_MEMPOOL_POLICY, not
         // TX_RECONSIDERABLE, because it cannot be bypassed using package validation.
         // This must be changed if package RBF is enabled.
+        if (ws.m_sibling_eviction_failure_string.has_value()) {
+            return state.Invalid(TxValidationResult::TX_MEMPOOL_POLICY, "v3-rule-violation and cannot bypass because insufficient fee",
+                                 strprintf("%s and %s", *ws.m_sibling_eviction_failure_string, *err_string));
+        }
         return state.Invalid(TxValidationResult::TX_MEMPOOL_POLICY, "insufficient fee", *err_string);
     }
 
     // Calculate all conflicting entries and enforce Rule #5.
     if (const auto err_string{GetEntriesForConflicts(tx, m_pool, ws.m_iters_conflicting, ws.m_all_conflicting)}) {
+        if (ws.m_sibling_eviction_failure_string.has_value()) {
+            return state.Invalid(TxValidationResult::TX_MEMPOOL_POLICY, "v3-rule-violation and cannot bypass because too many potential replacements",
+                                 strprintf("%s and %s", *ws.m_sibling_eviction_failure_string, *err_string));
+        }
         return state.Invalid(TxValidationResult::TX_MEMPOOL_POLICY,
                              "too many potential replacements", *err_string);
     }
     // Enforce Rule #2.
     if (const auto err_string{HasNoNewUnconfirmed(tx, m_pool, ws.m_iters_conflicting)}) {
+        if (ws.m_sibling_eviction_failure_string.has_value()) {
```

Comment:
This should be impossible right?  Maybe just an `Assume()` to check that there's no sibling eviction going on in this scenario?

Reply:
Added an `Assume`